promql/functions_test.go

```diff
@@ -86,3 +87,52 @@ func TestKahanSum(t *testing.T) {
 	expected := 2.0
 	require.Equal(t, expected, kahanSum(vals))
 }
+
+func TestMadOverTime(t *testing.T) {
```

Comment:
I would in fact not write tests in this way but only using the PromQL testing framework as suggested by @roidelapluie . (I still have a TODO item to remove all the very involved tests for histogram operations and convert them to the testing framework.)